src/Cake.Common/Build/MyGet/MyGetProvider.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Linq;
+using Cake.Core;
+
+namespace Cake.Common.Build.MyGet
+{
+    /// <summary>
+    /// Responsible for communicating with MyGet.
+    /// </summary>
+    public sealed class MyGetProvider : IMyGetProvider
+    {
+        private const string MessagePrefix = "##myget[";
+        private const string MessagePostfix = "]";
+        private static readonly Dictionary<string, string> _sanitizationTokens;
+        private readonly ICakeEnvironment _environment;
+
+        static MyGetProvider()
+        {
+            _sanitizationTokens = new Dictionary<string, string>
+            {
+                { "|", "||" },
+                { "\'", "|\'" },
+                { "\n", "|n" },
+                { "\r", "|r" },
+                { "[", "|['" },
+                { "]", "|]" }
+            };
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MyGetProvider"/> class.
+        /// </summary>
+        /// <param name="environment">The cake environment.</param>
+        public MyGetProvider(ICakeEnvironment environment)
+        {
+            if (environment == null)
+            {
+                throw new ArgumentNullException("environment");
+            }
+
+            _environment = environment;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the current build is running on MyGet.
+        /// </summary>
+        /// <value>
+        /// <c>true</c> if the current build is running on MyGet; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsRunningOnMyGet
+        {
+            get { return !string.IsNullOrWhiteSpace(_environment.GetEnvironmentVariable("BuildRunner")); }
```

Comment:
When running on MyGet, the `BuildRunner` environment variable should always have the value `MyGet`.
Perhaps change this line to something like this?

```csharp
get
{
  var buildRunner = _environment.GetEnvironmentVariable("BuildRunner");
  return !string.IsNullOrWhiteSpace(buildRunner) && string.Equals("MyGet", buildRunner, StringComparison.OrdinalIgnoreCase)
}
```